ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt

```diff
@@ -362,11 +363,21 @@ internal class PaywallViewModelImpl(
     private fun updateState() {
         viewModelScope.launch {
             try {
-                var currentOffering = options.offeringSelection.offering
-                if (currentOffering == null) {
-                    val offerings = purchases.awaitOfferings()
-                    currentOffering = options.offeringSelection.offeringIdentifier?.let { offerings[it] }
-                        ?: offerings.current
+                var currentOffering: Offering? = when (val offeringSelection = options.offeringSelection) {
+                    is OfferingSelection.OfferingType -> offeringSelection.offeringType
+                    is OfferingSelection.IdAndPresentedOfferingContext -> {
+                        val offerings = purchases.awaitOfferings()
+                        val presentedOfferingContext = offeringSelection.presentedOfferingContext
+                        val offering = options.offeringSelection.offeringIdentifier?.let { offerings[it] }
+                            ?: offerings.current
```

Comment:
Using `offeringSelection` would allow it to smart cast, avoiding the need to handle a nullable `offeringIdentifier`.
```suggestion
                        val offering = offerings[offeringSelection.offeringIdentifier] ?: offerings.current
```

Reply:
I think that doesn't work becuase `offeringSelection.offeringIdentifier` is nullable (Because it can be `OfferingSelection.None`).

Reply:
Oh yes yes, that works! Will do the change. Thanks!